.github/workflows/reusable-crypto3-testing-mac.yml

```diff
@@ -38,21 +43,47 @@ jobs:
           name: ${{ env.EVENT_FILE_ARTIFACT_NAME }}
           path: ${{ github.event_path }}
 
-
   build-and-test:
     name: "Build and Test using Matrix"
-    runs-on: ${{ matrix.os }}
+    runs-on: [macos-11]
     strategy:
-      # Set up a matrix to run the following 4 configurations:
-      # 1. <Linux, Release, latest G++ compiler toolchain on the default runner image, default generator>
-      # 2. <Linux, Release, latest Clang++ compiler toolchain on the default runner image, default generator>
-      # 3. <MacOS, Release, latest G++ compiler toolchain on the default runner image, default generator>
-      # 4. <MacOS, Release, latest Clang++ compiler toolchain on the default runner image, default generator>
+      # Set up a matrix to run the following 2 configurations:
+      # 1. <MacOS, Release, latest G++ compiler toolchain on the default runner image, default generator>
+      # 2. <MacOS, Release, latest Clang++ compiler toolchain on the default runner image, default generator>
       matrix:
-        os: [ubuntu-22.04, macos-12]
         cpp_compiler: [g++, clang++]
         build_type: [Release]
     steps:
+      - name: Install homebrew
+        run: >
+          /bin/bash -c "$(curl -fsSL
+            https://raw.githubusercontent.com/Homebrew/install/master/install.sh
+          )"
+
+      - name: Run brew install
+        id: brew-install
+        run: |
+          brew install \
+            boost@1.76 \
```

Comment:
Did you try _MarkusJx/install-boost@v2.4.1_ action to install boost?
Also, `boost-version` input becomes unused after changes

Reply:
I removed this one, left the MarkusJx/install-boost@v2.4.1.